python/TestHarness/testers/Tester.py

```diff
@@ -358,5 +359,10 @@ def checkRunnableBase(self, options, checks, test_list=None):
                 self.setStatus(reason, self.bucket_skip)
                 return False
 
+        # Check for display
+        if self.specs['display_required'] and os.environ['DISPLAY'] is None:
```

Comment:
Do macs always set the `DISPLAY` variable? It is set on mine but that might just be due to having Xquartz installed. We generally want these to run on macs.